### PR TITLE
Refactor Storage API Server

### PR DESCRIPTION
**Pull request by dotfloat**
This PR includes the following major changes:
- Conversion from Flask to FastAPI
- Removal of RdpApi and StorageApi classes, preferring to use SqlAlchemy
  without any additional wrappers. (It should be noted that the FastAPI
  cookiecutter project implements a bunch of CRUD functions that we
  could also implement. I decided not to because it's unnecessary at
  this point in time.)
- Addition of POST-ing endpoints
- Extraction now posts data via HTTP rather than directly via SqlAlchemy
- "New Storage/Old Plotting" has been removed. It was a proof-of-concept
  which we intend to reimplement within the Webviz framework (webviz-ert)

Resolves: #1102 
Resolves: #1116

### CODE REVIEW

File: ert_shared/storage/database_schema.py [hunk 1]
```diff
@@ -49,6 +49,14 @@ class Ensemble(Entity):
     parameter_definitions = relationship(
         "ParameterDefinition", back_populates="ensemble"
     )
+    responses = relationship(
+        "Response",
+        secondary="join(ResponseDefinition, Response, ResponseDefinition.id == Response.response_definition_id)",
+        # primaryjoin="Ensemble.id == Response.id",
```
Comment (oysteoh) [inline ert_shared/storage/database_schema.py:55]: should be removed / add additonal information why it should still be there to document something?

File: ert_shared/storage/app.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import os
+import sys
+from fastapi import FastAPI, Request
+from fastapi.exceptions import HTTPException
+from fastapi.exception_handlers import http_exception_handler
+from fastapi.responses import JSONResponse
+from fastapi.security import HTTPBasic
+
+from starlette import status
```
Comment (oysteoh) [inline ert_shared/storage/app.py:9]: starlette is not defined in project requirements

File: setup.py [hunk 3]
```diff
@@ -35,6 +35,7 @@ def package_files(directory):
         "console-progressbar==1.1.2",
         "decorator",
         "equinor-libres >= 8.0.0b0",
+        "fastapi",
         "flask",
```
Comment (oysteoh) [inline setup.py:39]: is flask necessary anymore?

File: setup.py [hunk 4]
```diff
@@ -35,6 +35,7 @@ def package_files(directory):
...
         "gunicorn",
```
Comment (oysteoh) [inline setup.py:40]: gunicorn necessary anymore?

**Review by oysteoh — APPROVED** [review]
I think this is awesome work! I've hammered it down with different cases and utilized the api from webviz ert - navigating back and forth - no exceptions and everything seems to work as i expect. It is a lot of code to read through, but it is clean - well-formed and quite easy to follow! If for some reason there is something wrong - it would be an easy thing from my understanding to pin it down and fix. I'm happy to approve!

File: ert_shared/storage/database_schema.py (hunk 1 above)
Comment (dotfloat) [reply ert_shared/storage/database_schema.py]: Removed. Was trying to define a join where there's an intermediary table, but I wasn't successful. I figured I'd rather remove the need for `ResponseDefinition` in a later PR.

File: ert_shared/storage/app.py (hunk 2 above)
Comment (dotfloat) [reply ert_shared/storage/app.py]: Changed to `from fastapi import status`, which exposes Starlette's `status`.

File: setup.py (hunk 3 above)
Comment (dotfloat) [reply setup.py]: No, removed.

File: setup.py (hunk 4 above)
Comment (dotfloat) [reply setup.py]: No, removed.

**Comment by dotfloat** [comment]
Jenkins test this please.

**Pull request merged**